### ISSUE TITLE
com.applovin.impl.sdk.s.a  java.lang.OutOfMemoryError

### ISSUE THREAD

**Issue by silenting** [AppLovin/AppLovin-MAX-SDK-Android#555]
### MAX SDK Version

11.11.3

### Device/Platform Info

android

### Current Behavior

**The current situation is that after updating our SDK, 4% of users have encountered this issue. Please resolve it for us.**
```
Exception java.lang.OutOfMemoryError: pthread_create (1040KB stack) failed: Try again
  at java.lang.Thread.nativeCreate
  at java.lang.Thread.start (Thread.java:887)
  at com.applovin.impl.sdk.s.a (s.java:135)
  at com.applovin.impl.sdk.s.a (s.java:111)
  at com.applovin.impl.sdk.s.a (s.java:90)
  at com.applovin.impl.mediation.g$11.run (g.java:850)
  at android.os.Handler.handleCallback (Handler.java:938)
  at android.os.Handler.dispatchMessage (Handler.java:99)
  at android.os.Looper.loop (Looper.java:223)
  at android.app.ActivityThread.main (ActivityThread.java:8001)
  at java.lang.reflect.Method.invoke
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:635)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:947)

```

```
Exception java.lang.OutOfMemoryError: pthread_create (1040KB stack) failed: Try again
  at java.lang.Thread.nativeCreate
  at java.lang.Thread.start (Thread.java:1063)
  at com.applovin.impl.sdk.s.a (s.java:135)
  at com.applovin.impl.sdk.s.a (s.java:111)
  at com.applovin.impl.sdk.s.a (s.java:90)
  at com.applovin.impl.mediation.g$11.run (g.java:850)
  at android.os.Handler.handleCallback (Handler.java:739)
  at android.os.Handler.dispatchMessage (Handler.java:95)
  at android.os.Looper.loop (Looper.java:148)
  at android.app.ActivityThread.main (ActivityThread.java:7406)
  at java.lang.reflect.Method.invoke
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1230)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1120)
```

### How to Reproduce

We were unable to reproduce this crash. The crash data was obtained from the Google Play backend.

**Comment by thomasmso** [comment]
Hi @silenting -

Our SDK is not the root cause of the problem. The crash shows that our SDK is trying to perform certain logic just when the device runs out of memory, so we can resolve that in the next SDK release this month and it should reveal the underlying. We'll keep you posted here for when that release occurs.

In the meantime, I recommend you check your app for any memory leaks if you're seeing a higher crash rate in the latest versions of your app. If the crash rate is the same as before, you can check individual crash reports to see if there are any `OutOfMemory` crashes that disappeared with the new update (I'd imagine those are the crashes that are now being represented by our SDK).

**Comment by silenting** [comment]
Okay, let's check again. Thank you very much. thomasmso，Thanks♪(･ω･)ﾉ

**Comment by thomasmso** [comment]
I can close issue when we release the SDK. Thanks!

**Comment by guowei0306** [comment]
Hi, we just released a new AppLovin SDK v12.0.0 which has this crash issue resolved.
